### ISSUE TITLE
find_syscall_table: didn't find do_syscall_64() on Ubuntu 5.15 kernel

### ISSUE THREAD

**Issue by ksap5** [Xilinx-CNS/onload#101]
Having trouble loading SFC drivers on 5.15 kernel under Ubuntu. Modprobe errors with: "ERROR: failed to find syscall table".  Drivers work fine on Ubuntu 20.04 with kernel 5.4 & 5.13

Am I missing something obvious?  

Installed via https://www.xilinx.com/content/dam/xilinx/publications/solarflare/onload/openonload/8-0-0-34/SF-120887-LS-9-OpenOnload-DKMS-Release-Package.zip

#Kernel version
```
$cat /proc/version_signature
Ubuntu 5.15.0-48.54~20.04.1-generic 5.15.53
```

#onload_tool reload output
```
$onload_tool reload
onload_tool: /sbin/modprobe -r sfc
onload_tool: /sbin/modprobe sfc
onload_tool: /sbin/modprobe onload
modprobe: ERROR: could not insert 'onload': Unknown symbol in module, or unknown parameter (see dmesg)
onload_tool: FAILED: /sbin/modprobe onload
```

#dmesg
```
[ 2228.362541] [sfc efrm] find_syscall_table: didn't find do_syscall_64()
[ 2228.362543] [sfc efrm] init_sfc_resource: ERROR: failed to find syscall table
```

#DKMS Installation output
```
Setting up linux-headers-5.15.0-48-generic (5.15.0-48.54~20.04.1) ...
/etc/kernel/header_postinst.d/dkms:
 * dkms: running auto installation service for kernel 5.15.0-48-generic

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...(bad exit status: 2)
/var/lib/dkms/onload/8.0.0.34/build/scripts/onload_install --newkernel 5.15.0-48-generic......................................
cleaning build area...(bad exit status: 2)

DKMS: build completed.

sfc.ko:
Running module version sanity check.
 - Original module
 - Installation
   - Installing to /lib/modules/5.15.0-48-generic/updates/dkms/

sfc_char.ko:
Running module version sanity check.
 - Original module
 - Installation
   - Installing to /lib/modules/5.15.0-48-generic/updates/dkms/

sfc_resource.ko:
Running module version sanity check.
 - Original module
 - Installation
   - Installing to /lib/modules/5.15.0-48-generic/updates/dkms/

onload.ko:
Running module version sanity check.

Good news! Module version 8.0.0.34 for onload.ko
exactly matches what is already found in kernel 5.15.0-48-generic.
DKMS will not replace this module.
You may override by specifying --force.

depmod...

Warning: Unable to find an initial ram disk that I know how to handle.
Will not try to make an initrd.

DKMS: install completed.
   ...done.
```

**Comment by abower-amd** [comment]
This repository is not the place to raise issues against supported releases of Onload: please contact support-nic@xilinx.com.

However, you may find that your issue matches one already raised here and is fixed in this repository, which you are welcome to try, but is community-supported only.

**Comment by frozenkold** [comment]
@ksap5 Hi, I am having a similar issue on a kernel version not officially supported by Onload. Were you able to fix the problem?

**Comment by ol-alexandra** [comment]
I've fixed this issue in this repository: see 673463c07a73768bb64e7f0d8149e7885f0c618e.

**Comment by frozenkold** [comment]
@ol-alexandra Thank you! I had just found where the entry to do_syscall_64 had changed in my kernel version, so that it no longer matches what Onload was searching for in that function. This is perfect :)
